lib/index.js

```diff
@@ -190,8 +192,11 @@ exports.register = function (server, options, next) {
 
     server.ext('onPreResponse', function (request, reply) {
 
+        var skipCacheStorage = !settings.errorOnCacheNotReady && !cache.isReady();
+
         if (!request.session ||
-            (!request.session._isModified && !request.session._isLazy)) {
+            skipCacheStorage ||
```

Comment:
I'm thinking this may not be placed correctly.  Seems to me we still want to go through the prepare() and cookie() functions.  We just want to skip the storage() function.  Thoughts?